great_expectations/core/expectation_validation_result.py

```diff
@@ -519,7 +519,7 @@ def __init__(  # noqa: PLR0913
     @property
     def asset_name(self) -> str | None:
         if "active_batch_definition" in self.meta:
-            return self.meta["active_batch_definition"].data_asset_name
+            return self.meta["active_batch_definition"]["data_asset_name"]
```

Comment:
```suggestion
            return self.meta["active_batch_definition"].get("data_asset_name")
```
Could we play it safe here and use `get` retrieval? We'd like to add some more rigid typing around meta but in the meanwhile, let's ensure we don't crash based on a missing value.